aws/data_source_aws_elasticache_replication_group.go

```diff
@@ -0,0 +1,112 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsElasticacheReplicationGroup() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsElasticacheReplicationGroupRead,
+		Schema: map[string]*schema.Schema{
+			"replication_group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"replication_group_description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"auth_token_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"automatic_failover_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"port": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"configuration_endpoint_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"primary_endpoint_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"number_cache_clusters": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"node_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"snapshot_window": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"snapshot_retention_limit": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsElasticacheReplicationGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+	input := &elasticache.DescribeReplicationGroupsInput{
+		ReplicationGroupId: aws.String(d.Get("replication_group_id").(string)),
+	}
+
+	resp, err := conn.DescribeReplicationGroups(input)
+	if err != nil {
+		return errwrap.Wrapf("Error retrieving Elasticache Replication Group: {{err}}", err)
+	}
+
+	var rg *elasticache.ReplicationGroup
+	for _, r := range resp.ReplicationGroups {
+		if *r.ReplicationGroupId == d.Get("replication_group_id").(string) {
+			rg = r
+		}
+	}
+	if rg == nil {
+		return fmt.Errorf("[ERROR] Elasticache Replication Group (%s) not found", d.Get("replication_group_id").(string))
```

Comment:
Nitpick: It's already clear that this is an error and it will be raised/displayed as such, there's no need to add an explicit `[ERROR]` here. It would be only valid in a log message.